src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -506,7 +507,7 @@ LOG_SUPER_DEBUG("Treating as the first box");
             {
 #pragma omp for
                 for (ct=0; ct<HII_TOT_NUM_PIXELS; ct++){
-                    previous_spin_temp->Tk_box[ct] = Tk_BC;
+                    previous_spin_temp->Tk_box[ct] = Tk_BC * (1.0 + cT_ad * perturbed_field->density[HII_R_INDEX(i,j,k)]*inverse_growth_factor_z*growth_factor_zp); //JBM QUESTION: is this properly defined here too?
```

Comment:
I think you need to replace `HII_R_INDEX(i,j,k)` with simply `ct` here. All these variables should be defined here, I'm pretty sure. Though probably @qyx268 will know better than I.